### ISSUE TITLE
Code to visualize the saliency map

### ISSUE THREAD

**Issue by SummerRainLeaf** [ispamm/PHBreast#1]
Hi, thanks for your work, can you provide the code to visualize the saliency map?

Thanks.

**Comment by elelo22** [comment]
Hi SummerRainLeaf,

We just uploaded the notebook Visualize.ipynb with the code to visualize the saliency map as well as the activation maps.

**Comment by SummerRainLeaf** [comment]
Hi, I have some questions. I tried to generate the saliency map and the activation map, but there are some mistakes.
Saliency map is empty, Act. map - Encoder is bad. Act. map - Classifier is Invalid shape () for image data.
Do you know why? Thanks.
![image](https://user-images.githubusercontent.com/112600735/190903425-2f063c14-dfa2-44be-8833-ca39cb653d65.png)

**Comment by elelo22** [comment]
Hi, I fixed an error in GetModel.py, maybe it was related to that? I just tested the notebook and it works. Are you using the same model (physenet) and model weights (physenet_inbreast4views_patches_pretr.pt) as in the notebook?

**Comment by SummerRainLeaf** [comment]
Yes, I'm using the same model (physenet) and model weights (physenet_inbreast4views_patches_pretr.pt) as in the notebook. After I changed GetModel.py it still doesn't produce the expected result. Thanks for the reply.

**Comment by elelo22** [comment]
Are you getting the same result as in the screenshot from 2 days ago? This is what I get if I comment out the line `net.load_state_dict(torch.load(weights, map_location='cpu'))`, which seems similar to yours, so it might be that in your case the weights are not being loaded correctly. 
![image](https://user-images.githubusercontent.com/60319356/191272161-9b4147db-a80f-408f-8bd4-5623e6368476.png)

Otherwise, sometimes Pillow doesn't read the images properly. I uploaded the exact same images (in Example_images) so at least we can rule out this problem (in any case check that the images are being loaded correctly, in my case Pillow reads them so that they are in [0, 65536], so I normalize them in [0, 255] dividing by 257). But if you're still gettting the same result from the screenshot, both these reasons don't explain the "invalid shape" of Act. map - Classifier, which doesn't happen when I run the notebook. Also download again the notebook now that I ran it again and uploaded it.

Let me know :)

**Comment by SummerRainLeaf** [comment]
Hello, I can get the result you gave correctly after re-downloading the file. 
I did a lot of experiments with the model you provided before, so I may have forgotten to change some parts of the code and not get the expected results. I made such a mistake because I‘m a junior, sorry for the inconvenience. 
Thanks for your continued replies.

**Comment by elelo22** [comment]
Hey, no problem at all! I'm glad it works now :)
